lib/file.js

```diff
@@ -210,22 +210,26 @@ internals.openStat = function (path, mode, callback) {
             }
 
             if (err.code === 'EACCES' || err.code === 'EPERM') {
+                return callback(Boom.forbidden(null, err.code));
+            }
+
+            if (path.indexOf('\u0000') !== -1) {      // handle "Path must be a string without null bytes" error
```

Comment:
If you are going to perform this check on every success case, why not move it up before the open operation which is more expensive?